src/NuGet.Core/NuGet.Protocol/README.md

```diff
@@ -1 +1,239 @@
-NuGet's implementation for interacting with feeds. Contains functionality for all feed types.
+# NuGet.Protocol
+
+NuGet.Protocol is a NuGet client SDK library that provides a set of APIs for interacting with NuGet feeds. It provides a way for developers to query NuGet feeds to discover packages and their dependencies, and also to download packages and their associated assets.
+
+# Usage
```

Comment:
> IMO the readme should be about a general overview and a few pointers instead of detailed code samples.

I highly recommend reading [Azure SDK's documentation guidelines](https://azure.github.io/azure-sdk/general_documentation.html). They've run extensive user studies to determine how to document their packages and READMEs. Their package READMEs contain code snippets that cover all common scenarios. For example: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/storage/Azure.Storage.Blobs#examples

In my mind NuGet's core scenarios are:

1. Find package
2. Download package
3. Upload package